### DISCUSSION TITLE
Using same form for data creation and update

### DISCUSSION THREAD

**Discussion by popeating** [react-hook-form/react-hook-form#9280] [Q&A]
I have a page with a list of item, a the end of the page there is a form (made with react hook form) to createa new item
Creating a new item is not a problem, validation is working and data is created and appended at the list.
When i click a item in the list  i would load it in the form (along with its id in a hidden field) so that it can be edited.
Clicking the item will set a the states (relatives to the field), the form is filled correctly with the state (i'm using defaultValue={state}), but when i submit the form, the data object submitted is empty, until i enter/leave every field
I tried using watch with no luck, is there a way to dinamically fill the form with states?

**Comment by leapful** [comment]
@popeating Please share a codesandbox url to re-produce the issue. Based on your description of the issue, the root cause might be related to dynamic setting states as value change on field element will not be updated to form values of react-hook-form.

You need to trigger "onChange" to set the value or use method "append" from hook "useFieldArray".
